docs/src/userguide/numpy_tutorial.rst

```diff
@@ -542,7 +638,23 @@ For MSVC (on Windows) you should use ``/openmp`` instead of ``-fopenmp``.
 The GIL must be released (see :ref:`Releasing the GIL <nogil>`), so this is why we
 declare our :func:`clip` function ``nogil``.
 
-.. literalinclude:: ../../examples/userguide/numpy_tutorial/compute_prange.pyx
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/userguide/numpy_tutorial/compute_prange.py
+            :lines: 3-
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/userguide/numpy_tutorial/compute_prange.pyx
+            :lines: 3-
+
+.. note::
+
+    Currently, Cython is checking whether there was a raised exception after every call of the function ``clip()``. Checking a raised exception requires GIL to be held which causes an overhead (see :issue:`5586` for the details).
```

Comment:
```suggestion
    Currently, Cython is checking whether there was a raised exception after every call of the function ``clip()``.
    Checking a raised exception requires the GIL to be held which causes overhead inside a `nogil` loop.
    The need to check here is a bug with functions returning a fused type (see :issue:`5586` for the details).
```

I've just made it clear that it applies to fused return types, since generally people would be better using `except -1`.